assets/js/blocks/cart-checkout/checkout/frontend.js

```diff
@@ -17,8 +20,28 @@ const getProps = () => {
 	};
 };
 
+const getErrorBoundaryProps = () => {
+	return {
+		header: __( 'Something went wrong…', 'woo-gutenberg-products-block' ),
+		text: __experimentalCreateInterpolateElement(
+			__(
+				'The checkout has encountered an unexpected error. <a>Try reloading the page</a>. If the error persists, please get in touch with us so we can assist.',
+				'woo-gutenberg-products-block'
+			),
+			{
+				a: (
+					// eslint-disable-next-line jsx-a11y/anchor-has-content
+					<a href="." />
```

Comment:
This is not working with permalinks of the type `http://one.wordpress.test/?p=123`. It redirects to the homepage.

Reply:
Updated with javascript. Take another look please?